.ci/jenkins_job_build.sh

```diff
@@ -107,6 +107,16 @@ then
 	fi
 fi
 
+# Check if we can fastpath return/skip the CI
+# Specifically do this **after** we have potentially done the static
+# checks, as we always want to run those.
+# Work around the 'set -e' dying if the check fails.
+{local ret; .ci/ci-fast-return.sh; ret=$?; } || true
```

Comment:
Shouldn't the `local ret` be *outside* this block since you are checking it's value outside of it?

Reply:
I did wonder that, when I lifted the code from:
https://github.com/kata-containers/tests/blob/master/.ci/kata-doc-to-script.sh#L186-L187
which, I think you wrote recently? Let's work out how best to get a retval without running into the `set -e`... I can't use the echo trick I used at https://github.com/kata-containers/tests/blob/master/lib/common.bash#L86, as the sub script being called generates output :-(. iirc, I did do some subshell thing like `$(set +e; xxx.sh)` before somewhere else... yeah, sometimes `set -e` makes our lives just a little more painful..

Reply:
I'm allowed to critique my own code then right? :smile:

Having checked the docs, I think this is correct (since `local` is for defining a function-local variable not a block-local one) but still looks odd.

Reply:
absolutely :-)  heh. Yep, looked odd when I copied it - but I trusted you'd done the right thing when you wrote it :-)  I'm more than happy to add a comment explaining how the oddness works... more than happy.